pennylane/typing.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file contains different PennyLane types."""
+import contextlib
+
+# pylint: disable=import-outside-toplevel, too-few-public-methods
+import sys
+from typing import Union
+
+import numpy as np
+from autograd.numpy.numpy_boxes import ArrayBox
+
+_TensorLike = Union[int, float, bool, complex, bytes, str, list, tuple, np.ndarray, ArrayBox]
```

Comment:
```suggestion
_TensorLike = Union[int, float, bool, complex, bytes, list, tuple, np.ndarray, ArrayBox]
```
Do we want to treat strings as tensorlike?

Reply:
Not really, I just added it because `numpy` treats it as array-like. But we can be a bit more strict.